Revise cross-cluster search docs per ES changes (#3818)
elastic/security-docs: docs/detections/rules-cross-cluster-search.asciidoc

This section explains the general process for setting up cross-cluster search in detection rules. For specific instructions on each part of the process, refer to the linked documentation.

> NOTE: This procedure uses TLS certificate authentication to add remote clusters. {stack} 8.10.0 introduces an alternate method using {ref}/remote-clusters-api-key.html[API key authentication], but it is not yet included in the following steps.

Reviewer: by mentioning 
>  but it is not yet included in the following steps.
do we imply this method is not supported/verified yet for cross-cluster rule runs?

FYI: I have raised support of it on out team meeting and we going to take action to find out who would be the best fir to verify rules cross-cluster behaviour

Author: > by mentioning
> 
> > but it is not yet included in the following steps.
> 
> do we imply this method is not supported/verified yet for cross-cluster rule runs?

@vitaliidm Yes, I wouldn't want to document anything more detailed until the team has verified this new method. Maybe this needs to be more explicit, something like "but it is not yet supported for detection rules"? I would update the statement, of course, once the team verifies the process.

Author: ```suggestion
NOTE: This procedure uses TLS certificate authentication to add remote clusters. {stack} 8.10.0 introduces an alternate method using {ref}/remote-clusters-api-key.html[API key authentication], but it is not yet supported for detection rules.
```

Reviewer: I think it is better and place us on a safe side, until it verified.
Also, according to https://www.elastic.co/guide/en/elasticsearch/reference/8.10/remote-clusters-api-key.html it is in `beta` stage. So we might rephrase, we strongly recommend the old way to set-up clusters, and cannot guarantee the new on would work. But either way is fine by me

> . On the local cluster, {ref}/remote-clusters-cert.html#remote-clusters-connect-cert[set up a connection] to the remote cluster. Note the unique name that you give to the remote cluster, because you'll need to use it throughout this process.

Reviewer 2: ```suggestion
. On the local cluster, {ref}/remote-clusters-cert.html[set up a connection] to the remote cluster. Note the unique name that you give to the remote cluster, because you'll need to use it throughout this process.
```

Reviewer 2: I would link to the top-level page for setting up remote clusters with certificates, otherwise users may skip the setting up trust part.

Author: @abdonpijpelink Good point, thanks. I was trying to avoid pushing the reader to another long page with another set of steps for them to follow, but it sounds like it's somewhat unavoidable. Maybe I can give some extra context to help the reader target what they really need on that other page:
```suggestion
. On the local cluster, {ref}/remote-clusters-cert.html[establish trust and set up a connection] to the remote cluster. Note the unique name that you give to the remote cluster, because you'll need to use it throughout this process.
```
